### ISSUE TITLE
New validation in Babel 7.7.0 breaks Metro

### ISSUE THREAD

**Issue by leonid-shevtsov** [facebook/metro#481]
Babel 7.7.0 will throw an error for an empty Identifier node: https://github.com/babel/babel/pull/10621

This breaks a specific place in Metro:
https://github.com/facebook/metro/blob/master/packages/metro/src/ModuleGraph/worker/JsFileWrapping.js#L81

```js
function functionFromProgram(
  program: Object,
  parameters: $ReadOnlyArray<string>,
): Object {
  return t.functionExpression(
    t.identifier(''),
    parameters.map(makeIdentifier),
    t.blockStatement(program.body, program.directives),
  );
}
```

Replacing the empty string with anything but empty resolves the issue.

I don't have enough familiarity with Metro to suggest a proper identifier. :)

Babel: 7.7.0
Metro: 0.57
Yarn: 1.19.1 

Metro config: 

```js
module.exports = {
  resolver: { sourceExts: ['js', 'jsx'] },
};
```

Babel config:

```json
{
  "presets": ["module:metro-react-native-babel-preset"],
  "plugins": [
    [
      "transform-inline-environment-variables",
      {
        "include": ["SENTRY_DSN"]
      }
    ]
  ]
}
```

**Comment by loganfsmyth** [comment]
Instead of `t.identifier('')`, Metro should be passing `null` as the identifier. Looks like this has been working by chance until now because Babel didn't have validation and no other plugins happened to trip over the empty identifier.

**Comment by nicolo-ribaudo** [comment]
We plan to re-introduce this validation in Babel 8 (https://github.com/babel/babel/issues/10746).

**Comment by MichaReiser** [comment]
Closing as this has been fixed in ed8a04f1dd59eaa85073251207c437d362b0457e
